queries/petrol.py

```diff
@@ -330,17 +334,17 @@ def _answ_for_petrol_query(q: Query, result) -> AnswerTuple:
             "Þar kostar bensínlítrinn {4} og dísel-lítrinn {5}."
         )
         dist_nf = distance_desc(station["distance"], case="nf")
-        dist_þf = distance_desc(station["distance"], case="þf")
+        dist_ef = distance_desc(station["distance"], case="ef", num_to_str=True)
         answer = answ_fmt.format(
             station["company"], station["name"], dist_nf, bensin_kr_desc, diesel_kr_desc
         )
         voice = voice_fmt.format(
             desc,
             station["company"],
             station["name"],
-            dist_þf,
-            bensin_kr_desc,
-            diesel_kr_desc,
+            dist_ef,
+            floats_to_text(bensin_kr_desc.replace(".", ""), gender="kvk"),
+            floats_to_text(diesel_kr_desc.replace(".", ""), gender="kvk"),
```

Comment:
Not entirely sure what's going on with the .replace(".", ""), could you explain?

Reply:
It was due to `iceformat_float` (called in `krona_desc`) adding thousand separators (e.g. "1.234,00") which messed with the conversion.